base/strings/io.jl

```diff
@@ -42,7 +42,7 @@ end
 function print(io::IO, xs...)
     lock(io)
     try
-        for x in xs
+        foreach(xs) do x
             print(io, x)
         end
```

Comment:
Could we use a generated function here if we are worried about closure overhead?

(Though I would hope that the compiler would inline and eliminate the closure when it unrolls the loop.)

Reply:
A generated function seems strictly worse since any argument to `print` not being concretely typed would always result in a dynamic dispatch. If we're worried about creating a closure we could use `Fix1` here, though I'd be surprised if this closure wasn't eliminated (for concrete types at least).

Reply:
Good point, it wouldn't hurt to do:
```jl
foreach(Fix1(print, io), xs)
```